h2o-core/src/main/java/water/util/ArrayUtils.java

```diff
@@ -1080,14 +1133,31 @@ public static void shuffleArray(int[] a, Random rng) {
   }
   public static double[] gaussianVector(int n) { return gaussianVector(n, System.currentTimeMillis()); }
   public static double[] gaussianVector(int n, long seed) { return gaussianVector(n, getRNG(seed)); }
-  public static double[] gaussianVector(int n, Random random) {
+  public static double[] gaussianVector(int n, Random random) { return gaussianVector(n, random, 0); }
+  public static double[] gaussianVector(int n, long seed, int zeroNum) {return gaussianVector(n, getRNG(seed), zeroNum); }
+  /**
+   * Make a new array initialized to random Gaussian N(0,1) values with the given seed.
+   * Make randomly selected {@code zeroNum} items zeros. Used in Extended isolation forest.
```

Comment:
imho this function shouldn't be here - it is too specific to your use case

instead of putting all the functionality into one function, keep the original gaussianVector

add a function that randomly zero-s out a portion of a row
and call them in EIF